lib/cloud_controller/upload_buildpack.rb

```diff
@@ -49,6 +55,27 @@ def upload_buildpack(buildpack, bits_file_path, new_filename)
 
     private
 
+    def raise_translated_api_error(buildpack)
+      if buildpack.errors.on([:name, :stack]).try(:include?, :unique)
+        raise CloudController::Errors::ApiError.new_from_details('BuildpackNameStackTaken', buildpack.name, buildpack.stack)
+      end
+      if buildpack.errors.on(:stack).try(:include?, :buildpack_cant_change_stacks)
+        raise CloudController::Errors::ApiError.new_from_details('BuildpackStacksDontMatch', buildpack.stack, buildpack.initial_value(:stack))
+      end
+      if buildpack.errors.on(:stack).try(:include?, :buildpack_stack_does_not_exist)
+        raise CloudController::Errors::ApiError.new_from_details('BuildpackStackDoesNotExist', buildpack.stack)
+      end
+    end
+
+    def determine_new_stack(buildpack, bits_file_path)
+      extracted_stack = Buildpacks::StackNameExtractor.extract_from_file(bits_file_path)
+      new_stack = [extracted_stack, buildpack.stack, Stack.default.name].find(&:present?)
+
+      new_stack
```

Comment:
There's no need to store the result of the `.find` in `new_stack`. Just return it.

With @ericpromislow

Reply:
:+1: Good spot. Updating code now...